### PR TITLE
[FLINK-14878]Support `use  catalog` through sqlUpdate() method in TableEnvironment

### PR DESCRIPTION
**Pull request by zjuwangg**
## What is the purpose of the change

*Support `USE CATALOG catalogName` through `sqlUpdate()` method in TableEnvironment*

## Brief change log

  - [c2375e0](https://github.com/apache/flink/commit/c2375e0a3af00cf91668375955d1caebef65f1fa) Add useCatalogOperation and support it both in flink/blink planner
  - [442d683](https://github.com/apache/flink/commit/442d683bfdc59204390b26d5433b42ed9a350cf0) add converter test and make sqlUseCatalog sql kind to other ddl to s
  - [ea05725](https://github.com/apache/flink/commit/ea0572560ca5e700a8d749484aac2edb17c07ba9) add related ITcase in CatalogTableITCase to both planner

## Verifying this change

This change added tests and can be verified as follows:
  - *CatalogTableITCase#testUseCatalog*
  - *SqlToOperationConverterTest#testUseCatalog*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: ( no )
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not applicable )

### CODE REVIEW

**Comment by zjuwangg** [comment]
cc @bowenli86 @xuefuz @lirui-apache @KurtYoung to have a review.

File: flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlUseCatalog.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
  */
 public class SqlUseCatalog extends SqlCall {
 
-	public static final SqlSpecialOperator OPERATOR = new SqlSpecialOperator("USE CATALOG", SqlKind.OTHER);
+	public static final SqlSpecialOperator OPERATOR = new SqlSpecialOperator("USE CATALOG", SqlKind.OTHER_DDL);
```
Comment (bowenli86) [inline flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlUseCatalog.java]: why is this required?

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java [hunk 2]
```diff
@@ -473,7 +474,11 @@ public void sqlUpdate(String stmt) {
...
 				dropTableOperation.isIfExists());
-		} else {
+		} else if (operation instanceof UseCatalogOperation) {
+			UseCatalogOperation useCatalogOperation = (UseCatalogOperation) operation;
+			catalogManager.setCurrentCatalog(useCatalogOperation.getCatalogName());
+		}
+		else {
 			throw new TableException(
 				"Unsupported SQL query! sqlUpdate() only accepts a single SQL statements of " +
 					"type INSERT, CREATE TABLE, DROP TABLE");
```
Comment (bowenli86) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java]: deos this need to be updated?

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/UseOperation.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.operations.ddl;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.operations.Operation;
+
+/**
+ * A {@link Operation} that describes the DDL statements, e.g. USE CATALOG or USE [catalogName.]dataBaseName.
```
Comment (bowenli86) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/UseOperation.java]:
```suggestion
 * An {@link Operation} that describes the DDL statements, e.g. USE CATALOG or USE [catalogName.]dataBaseName.
```

Comment (bowenli86) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/UseOperation.java]: and is "USE" counted as DDL?

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/internal/TableEnvImpl.scala [hunk 4]
```diff
@@ -29,18 +29,16 @@ import org.apache.flink.table.expressions.resolver.lookups.TableReferenceLookup
...
 import org.apache.flink.table.module.{Module, ModuleManager}
-import org.apache.flink.table.operations.ddl.{CreateTableOperation, DropTableOperation}
+import org.apache.flink.table.operations.ddl.{CreateTableOperation, DropTableOperation, UseCatalogOperation}
 import org.apache.flink.table.operations.utils.OperationTreeBuilder
 import org.apache.flink.table.operations.{CatalogQueryOperation, TableSourceQueryOperation, _}
 import org.apache.flink.table.planner.{ParserImpl, PlanningConfigurationBuilder}
 import org.apache.flink.table.sinks.{OverwritableTableSink, PartitionableTableSink, TableSink, TableSinkUtils}
 import org.apache.flink.table.sources.TableSource
 import org.apache.flink.table.util.JavaScalaConversionUtil
-
```
Comment (bowenli86) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/internal/TableEnvImpl.scala]: revert?

File: flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlUseCatalog.java (hunk 1 above)
Comment (zjuwangg) [reply flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlUseCatalog.java]: Both `validate(sqlNode: SqlNode)` methods in `FlinkPlannerImpl` in flink-planner/blink-planner will validate row type,  SqlKind.OTHER kind will failed the validation.

Comment (zjuwangg) [reply flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlUseCatalog.java]: After think twice, I revert the SqlKind and modify the planner validate condition.

**Comment by bowenli86** [comment]
sorry for the late comment, but I happened to notice that UseCatalogOperation resides in package `/table/operations/ddl/`. Since it's not ddl, should it just be relocated to `/table/operations/`?

**Comment by zjuwangg** [comment]
> sorry for the late comment, but I happened to notice that UseCatalogOperation and UseOperation reside in package `/table/operations/ddl/`. Since they are not ddl, should they be relocated to `/table/operations/`?

Make sense, addressed.
